File: model/user_company.go
package model

type UserCompanyPublic struct {
	ID
	TIMES

	CompanyID string `json:"company_id" gorm:"type:varchar(36)"`
	UserID    string `json:"user_id" gorm:"type:varchar(36)"`

	Company Company `json:"company"`
	User    User    `json:"user"`
}

type UserCompanyPrivate struct {
	Role string `json:"role" gorm:"size:100"`
}

type UserCompany struct {
	UserCompanyPublic
	UserCompanyPrivate
}

File: app/Models/UserCompany.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use App\Traits\SerializeDate;

class UserCompany extends Model
{
    use HasUuids, SerializeDate;

    /*
    |--------------------------------------------------------------------------
    | Variables
    |--------------------------------------------------------------------------
    */

    protected $table = 'user_companies';

    protected $fillable = [
        // REQUIRED
        'role',

        // FOREIGN KEY
        'company_id',
        'user_id',
    ];

    protected $hidden = [];


    /*
    |--------------------------------------------------------------------------
    | Relation - Belongs To
    |--------------------------------------------------------------------------
    */

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
